util/dvsim/FpvCfg.py

```diff
@@ -0,0 +1,213 @@
+        #   Proof,80%
+        # '''
+        results_str = "## " + self.results_title + "\n\n"
+        results_str += "### " + self.timestamp_long + "\n"
+        results_str += "### FPV Tool: " + self.tool.upper() + "\n"
+        results_str += "### LogFile dir: " + self.scratch_path + "/default \n\n"
+
+        colalign = ("center", ) * len(self.header)
+        table = [self.header]
+        self.result = {
+            "proven": 0,
+            "cex": 0,
+            "undetermined": 0,
+            "covered": 0,
+            "unreachable": 0,
+            "pass_rate": 0,
+            "cov_rate": 0
+        }
+        cov_header = ["stimuli", "coi", "proof"]
+        cov_table = [cov_header]
+        cov_colalign = ("center", ) * len(cov_header)
+
+        for mode in self.build_modes:
+            result_data = Path(subst_wildcards(self.build_dir, {"build_mode": mode.name}) +
+                               '/results.csv')
+            try:
+                with open(result_data, "r") as results_file:
+                    reader = csv.reader(results_file)
+                    result = dict((rows[0], rows[1]) for rows in reader)
+                    for key in result:
+                        self.result[result[key]] = self.result.get(result[key]) + 1
+                    total_tests = (self.result.get("proven") + self.result.get("cex") +
+                                   self.result.get("undetermined"))
+                    if total_tests == 0:
+                        self.result["pass_rate"] = "{0:.2f} %".format(0.00)
+                    else:
+                        pass_rate = ((self.result.get("proven") +
+                                      self.result.get("undetermined")) / total_tests * 100)
+                        self.result["pass_rate"] = "{0:.2f} %".format(round(pass_rate, 2))
+                    total_covs = self.result.get("covered") + self.result.get("unreachable")
+                    if total_covs == 0:
+                        self.result["cov_rate"] = "{0:.2f} %".format(0.00)
+                    else:
+                        cov_rate = self.result.get("covered") / total_covs * 100
+                        self.result["cov_rate"] = "{0:.2f} %".format(round(cov_rate, 2))
+
+                    table.append([
+                        self.name,
+                        self.result["proven"],
+                        self.result["cex"],
+                        self.result["undetermined"],
+                        self.result["covered"],
+                        self.result["unreachable"],
+                        self.result["pass_rate"],
+                        self.result["cov_rate"]])
+                os.remove(result_data)
+            except IOError as err:
+                log.error("%s", err)
+                results_str += str(err)
+
+            if (self.cov):
+                cov_data = Path(
+                    subst_wildcards(self.build_dir, {"build_mode": mode.name}) + "/cov.csv")
```

Comment:
This file will not exist if `--dry-run or -n` switch is passed, and an error will be thrown.

Reply:
Thanks for pointing this out. I fixed them from throwing error to warning.